### PR TITLE
fix(trino): fix `list_databases` to show all catalogs and `list_schemas` to allow the `database` argument

### PR DESCRIPTION
**Pull request by cpcloud**
This PR fixes the trino implementations of `list_databases` (only lists the current catalog instead of all available catalogs), and `list_schemas` (incorrect quoting of the schema in the base sqlalchemy implementation).

### CODE REVIEW

File: docker/trino/catalog/hive.properties [hunk 1]
```diff
@@ -1,6 +1,7 @@
 connector.name=hive
 
 hive.allow-drop-table=true
+hive.allow-rename-table=true
```
Comment (cpcloud) [inline docker/trino/catalog/hive.properties:4]: Moving this change to a separate PR

Comment (cpcloud) [reply docker/trino/catalog/hive.properties:4]: #6826

**Review by kszucs — APPROVED** [review]
